commitment/taproot.go

```diff
@@ -99,6 +99,34 @@ func NewPreimageFromLeaf(leaf txscript.TapLeaf) *TapscriptPreimage {
 	}
 }
 
+// ScriptFromLeafPreimage decodes the leafVersion and script from a taproot
+// leaf preimage.
+func ScriptFromLeafPreimage(preimage []byte) (*txscript.TapscriptLeafVersion,
+	[]byte, error) {
```

Comment:
I think we have an opportunity to use something other than `[]byte` here as the return type. It could be something like: `type Tapscript []byte`. Then we'd have `type TapscriptLeaf []byte` for the case that we expect the leaf version and length prefix to be there.